force-app/main/default/classes/LeadTriggerHandler.cls

```diff
@@ -31,14 +31,16 @@ public with sharing class LeadTriggerHandler {
 	 *   change the title to 'Assistant'.
 	 */
 	public static void handleTitleNormalization(List<Lead> leadsToNormalize) {
+		System.debug('RUN: handleTitleNormalization');
+		System.debug('leadsToNormalize: ' + leadsToNormalize);
 		for (Lead ld : leadsToNormalize) {
+			if (ld.title == null) {
+				continue;
+			}			
+			
 			if (ld.title == 'vp' || ld.title.contains('v.p.') || ld.title.contains('vice president')) {
 				ld.Title = 'Vice President';
-			} else if (
-				ld.title.contains('mgr') ||
-				ld.title.contains('manage') ||
-				ld.title.contains('head of department')
-			) {
+			} else if (ld.title.contains('mgr') || ld.title.contains('manage') || ld.title.contains('head of department')) {
```

Comment:
Try to use the containsIgnoreCase() method in case some of your titles are capitalized